src/viam/examples/modules/complex/CMakeLists.txt

```diff
@@ -154,21 +154,31 @@ install(
 # `test_complex_module.cpp` into the Viam C++ SDK testing suite.
 
 enable_testing()
-target_sources(viamsdk_test
+add_executable(test_complex_module test_complex_module.cpp)
+target_link_libraries(test_complex_module viamsdk_test)
+target_sources(test_complex_module
   PRIVATE
     gizmo/impl.cpp
-    gizmo/impl.hpp
     gizmo/api.cpp
-    gizmo/api.hpp
     summation/api.cpp
-    summation/api.hpp
     summation/impl.cpp
-    summation/impl.hpp
 
     ${MODULE_PROTO_OUTPUT_FILES}
 )
-target_include_directories(viamsdk_test
+target_include_directories(test_complex_module
   PUBLIC
     ${MODULE_PROTO_GEN_DIR}
 )
-viamcppsdk_add_boost_test(test_complex_module.cpp)
+
```

Comment:
What prevents you from continuing to use `viamcppsdk_add_boost_test` here, and then calling `target_sources`, `target_include_directories` and `target_link_libraries` on the resulting executable to add in what you need? I believe those calls are all addidative:

```
viamcppsdk_add_boost_test(test_complex_module.cpp)

target_sources(test_complex_module ...)
target_include_directories(test_complex_module ...)
target_link_libraries(test_complex_module...)
```

Also, I note that there are now three places where the generated sources are listed as `target_sources`: once in the module, once in the client, and once in the test.

But I think you can use the CMake "object libraries" to eliminate that: https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#object-libraries.

You would want to declare two: one for the client side sources and one for the server. Then you can name those in `target_link_libraries`.

I think you could even then use `target_include_directories` on the object library to attach `${MODULE_PROTO_GEN_DIR}` as `PUBLIC` and then it would transitively propagate to the libraries that link to the object libraries.

Can you give that a try?

Reply:
You did, I was imagining that the sources decomposed cleanly into client-y and server-y groups, but if that's not the case (and looking more closely I don't think it is), then there is no need for two `OBJECT` libraries and what you have done here is fine.